### PR TITLE
Fix release of waiting task with pre-init

### LINKED ISSUE #1857: cylc run --hold: prereqs caching breaks pre-initial dependencies
When this suite is run with `--hold`, then released, `foo` will be stuck in the waiting state:
```ini
[scheduling]
    initial cycle point = 2016-01-01
    final cycle point = 2016-01-02
    [[dependencies]]
        [[[PT6H]]]
            graph = "foo[-PT6H] => foo"
```

### PR DESCRIPTION
**Pull request by matthewrmshin**
On release of a waiting task, it would have its satifised dependencies reset, which meant that satisfied dependencies on pre-initial cycles were forgotten.

There is really no reason why we need to reset prerequisites, etc when a task is released.

Fix #1857.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
@hjoliver @benfitzpatrick please review.

File: lib/cylc/task_state.py [hunk 1]
```diff
@@ -360,7 +361,11 @@ def release(self):
         old_status = self._state_pre_hold
         self._state_pre_hold = None
         self.log(INFO, 'held => %s' % (old_status))
-        self.reset_state(old_status)
+
+        # Turn off submission and execution timeouts.
+        self.submission_timer_timeout = None
+        self.execution_timer_timeout = None
+        self.set_state(old_status)
```
Comment (matthewrmshin) [inline lib/cylc/task_state.py]: Actual fix here. I think we probably don't need to reset the timeouts, but it is harmless.

**Comment by hjoliver** [comment]
@matthewrmshin - as @arjclark noted here https://github.com/cylc/cylc/issues/1857#issuecomment-221261711 pre-initial ignore wipes out prerequisites, so they no longer exist in order for their satisified status to be forgotten.  The problem seems to be that prerequisite caching reports "all satisfied" as `False` even when there are no prerequisites left after pre-initial filtering, which is wrong.  So while I agree that your change is sufficient in this case, for completeness I think we should put @benfitzpatrick's change in as well: https://github.com/cylc/cylc/issues/1857#issuecomment-221269177.  Otherwise, this might come back to bite us again in other circumstances.

**Comment by matthewrmshin** [comment]
Branch rebased. Added prerequisite logic fixed by @benfitzpatrick.

File: tests/hold-release/19-no-reset-prereq-on-waiting/suite.rc [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            graph = """
+holder => t1
+holder => releaser
+t1[-P1Y] => t1
+"""
+[runtime]
+    [[t1]]
+        script = true
+    [[holder]]
+        script = cylc hold "${CYLC_SUITE_NAME}" 't1.2016'
```
Comment (benfitzpatrick) [inline tests/hold-release/19-no-reset-prereq-on-waiting/suite.rc]: Potential timing problem if hold doesn't stick before t1 fires off?

Comment (matthewrmshin) [inline tests/hold-release/19-no-reset-prereq-on-waiting/suite.rc]: Shouldn't do that after #1855, but I can put a `sleep 1` after this, just in case.

Comment (matthewrmshin) [inline tests/hold-release/19-no-reset-prereq-on-waiting/suite.rc]: Added `sleep 1`.

**Comment by arjclark** [comment]
I'm somewhat surprised we hadn't hit this problem until now - it's exposed by the pre-init stuff but it looks like it could get us in other circumstances

**Comment by hjoliver** [comment]
Review 1 - good.  Test battery passes.  Over to Ben...

**Pull request merged**